drivers/rtc/rtc_mchp_g1.c

```diff
@@ -0,0 +1,964 @@
+/*
+ * Copyright (c) 2025 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/clock_control/mchp_clock_control.h>
+#include "rtc_utils.h"
+
+#define DT_DRV_COMPAT microchip_rtc_g1
+
+LOG_MODULE_REGISTER(rtc_mchp_g1, CONFIG_RTC_LOG_LEVEL);
+
+#define RTC_MCHP_ALARM_1            (0)
+#define RTC_MCHP_ALARM_2            (1)
+#define RTC_TM_REFERENCE_YEAR       (1900U)
+#define RTC_REFERENCE_YEAR          (1996U)
+#define RTC_ADJUST_MONTH(month)     (month + 1U)
+#define RTC_ALARM_COUNT             DT_PROP(DT_NODELABEL(rtc), alarms_count)
+#define RTC_CALIB_PARTS_PER_BILLION (1000000000)
+#define RTC_CALIBRATE_PPB_MAX       (127)
+#define RTC_DATA_LOCK_INIT(p_lock)  k_mutex_init(p_lock)
+#define RTC_DATA_LOCK(p_lock)       k_mutex_lock(p_lock, K_FOREVER)
+#define RTC_DATA_UNLOCK(p_lock)     k_mutex_unlock(p_lock)
```

Comment:
These should not be defines, its just as short and clearer to just write the actual line

Reply:
Also, use semaphores for locking, mutexes are only needed if tracking thread ownership is needed, like for condvar